internal/services/runservice/api/api.go

```diff
@@ -313,6 +313,110 @@ func sendLogs(w http.ResponseWriter, r io.Reader) error {
 	}
 }
 
+type LogsDeleteHandler struct {
+	log *zap.SugaredLogger
+	e   *etcd.Store
+	ost *objectstorage.ObjStorage
+	dm  *datamanager.DataManager
+}
+
+func NewLogsDeleteHandler(logger *zap.Logger, e *etcd.Store, ost *objectstorage.ObjStorage, dm *datamanager.DataManager) *LogsDeleteHandler {
+	return &LogsDeleteHandler{
+		log: logger.Sugar(),
+		e:   e,
+		ost: ost,
+		dm:  dm,
+	}
+}
+
+func (h *LogsDeleteHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	q := r.URL.Query()
+
+	runID := q.Get("runid")
+	if runID == "" {
+		httpError(w, util.NewErrBadRequest(errors.Errorf("runid is empty")))
+		return
+	}
+	taskID := q.Get("taskid")
+	if taskID == "" {
+		httpError(w, util.NewErrBadRequest(errors.Errorf("taskid is empty")))
+		return
+	}
+
+	_, setup := q["setup"]
+	stepStr := q.Get("step")
+	if !setup && stepStr == "" {
+		httpError(w, util.NewErrBadRequest(errors.Errorf("setup is false and step is empty")))
+		return
+	}
+	if setup && stepStr != "" {
+		httpError(w, util.NewErrBadRequest(errors.Errorf("setup is true and step is %s", stepStr)))
+		return
+	}
+
+	var step int
+	if stepStr != "" {
+		var err error
+		step, err = strconv.Atoi(stepStr)
+		if err != nil {
+			httpError(w, util.NewErrBadRequest(errors.Errorf("step %s is not a valid number", stepStr)))
+			return
+		}
+	}
+
+	if err, deleteError := h.deleteTaskLogs(ctx, runID, taskID, setup, step, w); err != nil {
+		h.log.Errorf("err: %+v", err)
+		if deleteError {
+			switch err.(type) {
+			case *util.ErrNotExist:
+				httpError(w, util.NewErrNotExist(err))
+			case *util.ErrBadRequest:
+				httpError(w, util.NewErrBadRequest(err))
+			default:
+				httpError(w, util.NewErrInternal(err))
+			}
+		}
+	}
+}
+
+func (h *LogsDeleteHandler) deleteTaskLogs(ctx context.Context, runID, taskID string, setup bool, step int, w http.ResponseWriter) (error, bool) {
+	r, err := store.GetRunEtcdOrOST(ctx, h.e, h.dm, runID)
```

Comment:
The returned bool isn't need, it is used in `readTaskLogs` because we have to know if the errors happened before or after writing the response header (since we cannot send another response header when we already started streaming the body)